### ISSUE TITLE
Set Half Wave Viewing Mode as default

### ISSUE THREAD

**Issue by skynetradio** [audacity/audacity#7261]
### Your idea

Set Halfwave Mode as standard view

### Problem to be solved

Unnecessary views are hidden, better use of the monitor space, larger display of the recorded radio contribution.

Here is a screenshot of what it should look like.
![Zwischenablage03](https://github.com/user-attachments/assets/147c25a0-37cf-4518-b8c3-03334b459b7a)

If the selection was simply added here it would be perfect!!
![Unbenannt](https://github.com/user-attachments/assets/93cdaf97-b7ef-48a2-9e47-1e08e02a8f0a)

### Additional context

The reason is that I use Audacity for voice tracking in the radio sector and it is easier for me to quickly edit a record.

**Comment by skynetradio** [comment]
I wanted to mention something else, it seems that the view is not exactly separated. It would be nice if you could do that at the same time and everything below 0.0dB is not displayed, you can still see a little bit of the other half view.

Regards Tom

![Zwischenablage04](https://github.com/user-attachments/assets/590acd22-cf87-420e-83c3-160aff9c3d23)

**Comment by petersampsonaudacity** [comment]
I _(and I suspect many, maybe most, other users)_ would _**not**_ want this to be the default setting.

For my money full waveform should remain the default setting.

**Comment by SteveDaulton** [comment]
> For my money full waveform should remain the default setting.

I strongly agree, and this is backed up by users that voiced an opinion the last time this was discussed.

**Comment by skynetradio** [comment]
> > For my money full waveform should remain the default setting.
> 
> I strongly agree, and this is backed up by users that voiced an opinion the last time this was discussed.

It's clear that the full waveform should remain standard, I just want there to be an option to change it, because that would be interesting and important for me.

**Comment by skynetradio** [comment]
It's important to me that you understand that I don't want to force anything on anyone. I think we're misunderstanding each other. I would just like to be able to choose it and nothing else. The standard should of course be the full waveform.

**Comment by skynetradio** [comment]
Oh no, definitely not like they said!!
Maybe my title isn't very good, sorry.
I'll change that.

> > It's important to me that you understand that I don't want to force anything on anyone.
> 
> @skynetradio
> 
> But you clearly wrote
> 
> > Set Halfwave Mode as standard view
> 
> That to me, and @SteveDaulton . means make it the app standard default.
> 
> I have no objection to it being a personal default that you should be able to set - but it certainly should not be the OOTB factory setting.

Oh no, definitely not like they said!!
Maybe my title isn't very good, sorry.
I'll change that.

**Comment by Kurtsley** [comment]
I'm no authority on this, but it seems you are talking about two separate things. The view modes choose how you view the audio, waveform vs spectrogram. I think you may be talking about adding a default zoom mode since halfwave is a zoom setting. Maybe halfwave could be added as a convenient keyboard shortcut like the other zoom modes? It might also be possible to include it in the Zoom Toggle menu, but I'm not sure that's appropriate.

As for the second issue, changing line 170 to 0.0f seems to work.

https://github.com/audacity/audacity/blob/eefa92b8fc94cd9138300df6d32390d91170a818/src/tracks/playabletrack/wavetrack/ui/WaveformVZoomHandle.cpp#L169-L172

There must have been a reason that it is set the way that it is, but I'm not too familiar with this program. Thoughts?

**Comment by petersampsonaudacity** [comment]
> Maybe halfwave could be added as a convenient keyboard shortcut like the other zoom modes? It might also be possible to include it in the Zoom Toggle menu,

Adding say a command **View > Zoom > Half Wave** would enable a user to set a custom keyboard shortcut for this,

Or it could be a "hidden" command by adding it somewhere in the **Extra** menus, but I can't see a suitable place for that.  But that would still enable the command to be seen when setting **Shortcuts** preferences.  _A recent example of this is **Export Selected Audio** which was added back (following user feedback requesting it) to the Extra > Export menu, mainly just so a shortcut could be set for it._

**Comment by skynetradio** [comment]
As you mentioned, petersampsonaudacity, this selection shown in your screenshot (green) would be the best solution. With one click, Audacity works the whole time in half wave mode.

**Comment by Kurtsley** [comment]
@petersampsonaudacity

I seem to have this working correctly, but I think it's a bit of a hack.

If the doSetMinimized function is changed to zoom to halfwave when the collapsetohalfwave preference is set, everything appears to work. Since this gets called so often, this essentially accomplishes all the goals of the enhancement request.
https://github.com/audacity/audacity/blob/2fa42e2c9afa6c248a53574542ed50d1fe15b062/src/tracks/playabletrack/wavetrack/ui/WaveformView.cpp#L138-L145
Changed to:
```
   if (wt &&  bHalfWave) {
      auto &cache = WaveformScale::Get(*wt);
      // Zoom to show fractionally more than the top half of the wave.
      cache.SetDisplayBounds( -0.01f, 1.0f );
   }
```
*However* I would rather change the default zoom level that the track is created with. This seems to be an issue with zooming as changing the default bounds locks the track so the zoom can't be reset. In the end, this current fix seems to work as intended.

Secondly, should the preference name in the cfg file be changed from collapseToHalfWave to something more appropriate? I assume changing config names could be a problem with existing users.
